File: AM_Program/AM_Shared/IToolRunner.vb
'*********************************************************************************************************
' Written by Dave Clark for the US Department of Energy 
' Pacific Northwest National Laboratory, Richland, WA
' Copyright 2006, Battelle Memorial Institute
' Created 06/07/2006
'
' Last modified 07/29/2008
'*********************************************************************************************************

Option Strict On

Public Interface IToolRunner

	'*********************************************************************************************************
	'Insert general class description here
	'*********************************************************************************************************

#Region "Properties"
	ReadOnly Property EvalCode As Integer
	ReadOnly Property EvalMessage As String

	ReadOnly Property ResFolderName() As String

	' Explanation of what happened to last operation this class performed
	' Used to report error messages
	ReadOnly Property Message() As String

	ReadOnly Property NeedToAbortProcessing() As Boolean

	' the state of completion of the job (as a percentage)
	ReadOnly Property Progress() As Single
#End Region

#Region "Methods"
    Sub Setup(mgrParams As IMgrParams, jobParams As IJobParams, StatusTools As IStatusFile, SummaryFile As clsSummaryFile)

	Function RunTool() As IJobParams.CloseOutType

#End Region

End Interface

File: AM_Program/AM_Shared/IToolRunner.cs

//*********************************************************************************************************
// Written by Dave Clark for the US Department of Energy
// Pacific Northwest National Laboratory, Richland, WA
// Copyright 2006, Battelle Memorial Institute
// Created 06/07/2006
//
//*********************************************************************************************************

namespace AnalysisManagerBase
{

    public interface IToolRunner
    {

        #region "Properties"

        /// <summary>
        /// Evaluation code to be reported to the DMS_Pipeline DB
        /// </summary>
        int EvalCode { get; }

        /// <summary>
        /// Evaluation message to be reported to the DMS_Pipeline DB
        /// </summary>
        string EvalMessage { get; }

        /// <summary>
        /// Publicly accessible results folder name and path
        /// </summary>
        string ResFolderName { get; }

        /// <summary>
        /// Explanation of what happened to last operation this class performed
        /// </summary>
        string Message { get; }

        /// <summary>
        /// Set this to true if we need to abort processing as soon as possible due to a critical error
        /// </summary>
        bool NeedToAbortProcessing { get; }

        #endregion

        /// <summary>
        /// The state of completion of the job (as a percentage)
        /// </summary>
        float Progress { get; }

        #region "Methods"

        /// <summary>
        /// Initializes class
        /// </summary>
        /// <param name="stepToolName">Name of the current step tool</param>
        /// <param name="mgrParams">Object holding manager parameters</param>
        /// <param name="jobParams">Object holding job parameters</param>
        /// <param name="statusTools">Object for status reporting</param>
        /// <param name="summaryFile">Object for creating an analysis job summary file</param>
        /// <param name="myEMSLUtilities">MyEMSL download Utilities</param>
        /// <remarks></remarks>
        void Setup(
            string stepToolName,
            IMgrParams mgrParams,
            IJobParams jobParams,
            IStatusFile statusTools,
            clsSummaryFile summaryFile,
            clsMyEMSLUtilities myEMSLUtilities);

        /// <summary>
        /// Runs the analysis tool
        /// Major work is performed by overrides
        /// </summary>
        /// <returns>CloseoutType enum representing completion status</returns>
        /// <remarks></remarks>
        CloseOutType RunTool();

        #endregion

    }

}